src/Symfony/Component/Console/Tests/Command/InvokableCommandTest.php

```diff
@@ -0,0 +1,152 @@
+        self::assertSame('verbose', $verboseInputOption->getName());
+        self::assertSame('v', $verboseInputOption->getShortcut());
+        self::assertFalse($verboseInputOption->isValueRequired());
+        self::assertTrue($verboseInputOption->isNegatable());
+        self::assertNull($verboseInputOption->getDefault());
+
+        $groupsInputOption = $command->getDefinition()->getOption('groups');
+        self::assertSame('groups', $groupsInputOption->getName());
+        self::assertTrue($groupsInputOption->isArray());
+        self::assertSame('User groups', $groupsInputOption->getDescription());
+        self::assertSame([], $groupsInputOption->getDefault());
+
+        $rolesInputOption = $command->getDefinition()->getOption('roles');
+        self::assertSame('roles', $rolesInputOption->getName());
+        self::assertFalse($rolesInputOption->isValueRequired());
+        self::assertTrue($rolesInputOption->isArray());
+        self::assertSame(['ROLE_USER'], $rolesInputOption->getDefault());
+        self::assertTrue($rolesInputOption->hasCompletion());
+        $rolesInputOption->complete(new CompletionInput(), $suggestions = new CompletionSuggestions());
+        self::assertSame(['ROLE_ADMIN', 'ROLE_USER'], array_map(static fn (Suggestion $s) => $s->getValue(), $suggestions->getValueSuggestions()));
+    }
+
+    public function testInvalidArgumentType()
+    {
+        $command = new Command('foo');
+        $command->setCode(function (#[Argument] object $any) {});
+
+        $this->expectException(LogicException::class);
+        $this->expectExceptionMessage('The type "object" of parameter "$any" is not supported as a command argument. Only "string", "bool", "int", "float", "array" types are allowed.');
+
+        $command->getDefinition();
+    }
+
+    public function testInvalidOptionType()
+    {
+        $command = new Command('foo');
+        $command->setCode(function (#[Option] object $any) {});
+
+        $this->expectException(LogicException::class);
+        $this->expectExceptionMessage('The type "object" of parameter "$any" is not supported as a command option. Only "string", "bool", "int", "float", "array" types are allowed.');
+
+        $command->getDefinition();
+    }
+
+    public function testInvalidReturnValueType()
+    {
+        $command = new Command('foo');
+        $command->setCode(new class {
+            public function __invoke()
+            {
+                return 'bar';
+            }
+        });
+
+        $this->expectException(LogicException::class);
+        $this->expectExceptionMessage('The command "foo" must return either void or an integer value in the "__invoke" method, but "string" was returned.');
+
+        $command->run(new ArrayInput([]), new NullOutput());
+    }
+
+    public function getSuggestedRoles(CompletionInput $input): array
+    {
+        return [
+            new Suggestion('ROLE_ADMIN'),
+            new Suggestion('ROLE_USER'),
+        ];
+    }
```

Comment:
Wrapping the values in a `Suggestion` should not be necessary. Is it to test something?
```suggestion
        return ['ROLE_ADMIN', 'ROLE_USER'];
    }
```